### PR TITLE
CORTX-26260: HARE: Support Config SW Upgrade

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
Tested below command after doing 3 node deployment

> /opt/seagate/cortx/hare/bin/hare_setup upgrade --changeset yaml:///etc/cortx/changeset.conf --config yaml:///etc/cortx/cluster.conf

Signed-off-by: Swapnil Gaonkar <swapnil.gaonkar@seagate.com>

### CODE REVIEW

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -1279,12 +1291,13 @@ def create_parser() -> argparse.ArgumentParser:
                        help_str='Performs the Hare rpm post-upgrade tasks',
                        handler_fn=noop))
 
-    add_service_argument(
```
Comment (vaibhavparatwar) [inline provisioning/miniprov/hare_mp/main.py:1282]: Is this no more required now?

Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py:1282]: It is there, we are just adding one more layer on top of it to support changeset argument.

**Comment by pavankrishnat** [comment]
@SwapnilGaonkar7 3N deployment is successful 
```
kubectl exec -it cortx-data-ssc-vm-g4-rhev4-1592-699c8fd6dc-wnfn2 -c cortx-hax -- /bin/bash
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1592 /]# /opt/seagate/cortx/hare/bin/hare_setup upgrade --config yaml:///etc/cortx/cluster.conf
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1592 /]# echo $?
0
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1592 /]# rpm -qa | grep cortx
cortx-provisioner-2.0.0-5044_1c0926ec.noarch
cortx-hare-2.0.0-6801_git819b99d.el8.x86_64
cortx-motr-2.0.0-6801_gitf2ea7736.el8.x86_64
cortx-py-utils-2.0.0-6801_1e91d2e3.noarch
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1592 /]#
```
```
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1590 /]# /opt/seagate/cortx/hare/bin/hare_setup upgrade --changeset yaml:///etc/cortx/changeset.conf
usage: hare_setup upgrade [-h] --config CONFIG [--services SERVICES]
                          [--changeset CHANGESET]
hare_setup upgrade: error: the following arguments are required: --config/-c
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1590 /]# /opt/seagate/cortx/hare/bin/hare_setup upgrade --config yaml:///etc/cortx/cluster.conf
[root@cortx-data-headless-svc-ssc-vm-g4-rhev4-1590 /]# echo $?
0
```

**Review by mssawant — APPROVED** [review]

**Review by vaibhavparatwar — APPROVED** [review]

**Comment by vaibhavparatwar** [comment]
retest this please

**Pull request merged**